Controller/Adminhtml/Slide/Delete.php

```diff
@@ -24,27 +47,28 @@ protected function _isAllowed()
      */
     public function execute()
     {
-        $id = $this->getRequest()->getParam('slide_id');
         /** @var \Magento\Backend\Model\View\Result\Redirect $resultRedirect */
         $resultRedirect = $this->resultRedirectFactory->create();
+        $id = $this->getRequest()->getParam('slide_id');
+
         if (!$id) {
             // display error message
-            $this->messageManager->addError(__('We can\'t find slide to delete.'));
+            $this->messageManager->addErrorMessage(__('We can\'t find slide to delete.'));
             // go to grid
             return $resultRedirect->setPath('*/*/');
         }
 
         try {
-            /* @var \Scandiweb\Slider\Model\Slide $model */
-            $model = $this->_objectManager->create('Scandiweb\Slider\Model\Slide');
-            $model->load($id);
-            $model->delete();
-
-            $this->messageManager->addSuccess(__('The slide has been deleted.'));
+            $slide = $this->slideRepository->get($id);
+            $this->slideRepository->delete($slide);
```

Comment:
Might be worth to implement slider repository `deleteById` method

Reply:
It's not gonna help here, because we need slider ID from slide.